common:
  lid_topic: "/livox/lidar"
  imu_topic: "/new_imu"
  # imu_topic: "/camera/imu"
  time_sync_en: true         # ONLY turn on when external time synchronization is really not possible

preprocess:
  lidar_type: 1                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
  scan_line: 1
  blind: 0.1
  time_scale: 1e-3

noise_model:
  ranging_cov: 0.02 #0.04 0.02
  angle_cov: 0.1 #0.1 0.05

mapping:
#  acc_cov: 0.1
#  gyr_cov: 0.003
  acc_cov_scale: 0.1 #0.4 #0.1 #100
  gyr_cov_scale: 0.1 #0.25 #0.003 #10000
  b_acc_cov: 0.001 # 0.0043 0.001 # 0.001 #0.0001 #0.001
  b_gyr_cov: 0.001 # 0.000266 0.00003 # 0.00003 #0.0001 #0.00003
  fov_degree: 70
  det_range: 260.0
  extrinsic_est_en: false      # true: enable the online estimation of IMU-LiDAR extrinsic
  # extrinsic_T: [ -0.017020, 0.085815, -0.024827 ]
  # extrinsic_R: [ 0.046790, -0.998811, 0.013708,
  #                  -0.023212, -0.014807, -0.999621,
  #                  0.998635, 0.046454, -0.023877]
  extrinsic_T: [ -0.0275814, 0.0145758, -0.085423 ]
  extrinsic_R: [ 1, 0, 0,
                 0, 1, 0,
                 0, 0, 1]

camera:
    # img_topic: /usb_cam/image_raw
    # img_topic:  /camera/image_color
    img_topic: "/camera/color/image_raw"
    # img_topic: "/color/image"
    #xiyuan
    # lidar to camera
    Rcl: [0.0110805,-0.999823,0.0151929,
         -0.0410198,-0.0156355,-0.999036,
          0.999097,0.0104466,-0.0411858]
    Pcl: [0.0183, 0.0762623, -0.0305996]
    # camera to lidar
    # Rcl: [0.0110805, -0.0410198, 0.999097,
    #      -0.999823, -0.0156355, 0.0104466,
    #       0.0151929, -0.999036, -0.0411858]
    # Pcl: [0.0334975, 0.0198088, 0.0746505]

publish:
  path_publish_en: true
  scan_publish_en: true       # false: close all the point cloud output
  scan_effect_pub_en: true    # true: publish the pointscloud of effect point
  dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
  scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame
  # voxel
  pub_voxel_map: false
  publish_max_voxel_layer: 0         # 不再使用
  # publish_only_voxel_layer: 3        # only publish 0,1,2 layer's plane
  pub_point_cloud: false
  dense_map_enable: false
  pub_point_cloud_skip: 1             # publish one points per five points

path_save_en: false                 # 保存轨迹，用于精度计算和比较

pcd_save:
  pcd_save_en: false
  interval: -1                 # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

feature_extract_enable: false
point_filter_num: 3
max_iteration: 4 #10 default 3
dense_map_enable : 1
depth_img_enable : 1
filter_size_surf: 0.4 #0.05
filter_size_map: 0.1 #0.15            # 暂时未用到，代码中为0， 即倾向于将降采样后的scan中的所有点加入map
cube_side_length: 20 #20 default 1000 ivox参数

debug : 0
grid_size : 40 #30 40
patch_size : 8 #6 8
img_enable : 1
lidar_enable : 1
outlier_threshold : 600 # 78 100 156 建议较暗的场景为50~250，较亮的场景为500~1000。该值越小，vio子模块越快，但抗退化能力越弱。
ncc_en: false 
ncc_thre: 0 
img_point_cov : 1 # 100 The covariance of photometric errors per pixel.
laser_point_cov : 0.001 # 0.001 The covariance of point-to-plane redisual per point.
cam_fx: 426.551167166986
cam_fy: 426.6142447926181
cam_cx: 429.1314081919893
cam_cy: 247.64043896840604

ivox_grid_resolution: 0.2        # default=0.2
ivox_nearby_type: 18             # 6, 18, 26
esti_plane_threshold: 0.1        # default=0.1

voxel:
    voxel_size: 2.0
    max_layer: 4                 # 4 layer, 0, 1, 2, 3
    layer_point_size: [5, 5, 5, 5, 5]
    plannar_threshold: 0.01
    max_points_size: 1000
    max_cov_points_size: 1000
    
Result:
    write_kitti_log: false
    result_path: "/home/hr/workspace/faster_lio_ws/src/faster-lio/kitt_log.txt"
